python/tvm/relay/frontend/pytorch.py

```diff
@@ -1875,7 +1875,9 @@ def Bool(self, inputs, input_types):
 
     def Float(self, inputs, input_types):
         assert len(inputs) == 1
-        return _op.cast(inputs[0], "float32")
+        if isinstance(inputs[0], _expr.Expr):
+            return inputs[0]
+        return float(inputs[0])
```

Comment:
OK, Previous implement is
 `return _op.cast(inputs[0], "float32")
`
If we set inputs[0] as a integer, for example 100, _op.cast(100, "float32") will crash. 
we can do the same thing in test_forward with function convert_i, that is why we need think it as a bug.

Reply:
We should probably check the dtype of inputs and add cast if necessary.

Reply:
This modification is inspired by
```
    def int(self, inputs, input_types):
        if isinstance(inputs[0], _expr.Expr):
            return inputs[0]
        return int(inputs[0])
```
We think self.Int is very likely self.Float.

Reply:
Restore from previous version, it is not a bug under trace mode.

Reply:
> @delldu Please drop support for script mode. Just because your model has control flow doesn't mean these ops also need to be scripted. Remember, everyone can use these op conversion. You can do trace and selectively script certain parts of your model using `torch.jit._script_if_tracing`. This is how MaskRCNN in torchvision is implemented, for example. See https://github.com/pytorch/vision/blob/master/torchvision/models/detection/roi_heads.py#L454

Thanks a lot. It is good reference.

Reply:
Support script mode has been droped.